hs_core/page_processors.py

```diff
@@ -68,19 +69,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
                     if is_authorized:
                         tool_url = tool_res_obj.metadata.url_bases.first().value \
                             if tool_res_obj.metadata.url_bases.first() else None
-                        u = user.username if user.is_authenticated() else "anonymous"
-                        if tool_url:
-                            if tool_url.endswith('/'):
-                                tool_url = tool_url[:-1]
-                        else:
-                            tool_url = ""
                         tool_icon_url = tool_res_obj.metadata.tool_icon.first().url \
                             if tool_res_obj.metadata.tool_icon.first() else "raise-img-error"
-                        tl = {'title': str(tool_res_obj.metadata.title.value),
-                              'icon_url': tool_icon_url,
-                              'url': "{0}{1}{2}{3}{4}{5}".format(tool_url, "/?res_id=", content_model.short_id,
-                                                                 "&usr=", u, "&src=hs")}
-                        relevant_tools.append(tl)
+                        hs_term_dict_user = {}
+                        hs_term_dict_user["HS_USR_NAME"] = request.user.username if request.user.is_authenticated() else "anonymous"
+                        tool_url_new = parse_app_url_template(tool_url, [content_model.get_hs_term_dict(), hs_term_dict_user])
+                        if tool_url_new is not None:
```

Comment:
@selimnairb  currently we just not display that webapp on other resource landing pages